source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -204,6 +207,11 @@ FilterStatus ConnectionManager::ResponseDecoder::transportEnd() {
 
   ConnectionManager& cm = parent_.parent_;
 
+  if (cm.read_callbacks_->connection().state() == Network::Connection::State::Closed) {
+    complete_ = true;
+    throw EnvoyException("thrift downstream gone");
```

Comment:
Can we improve the message?  Perhaps `downstream connection is closed`. This will come out in log as ` thrift response error: downstream connection is closed`